toolz/itertoolz.py

```diff
@@ -1055,3 +1056,14 @@ def random_sample(prob, seq, random_state=None):
 
         random_state = Random(random_state)
     return filter(lambda _: random_state.random() < prob, seq)
+
+
+def flat(level, seq):
+    """ Flatten a possible nested sequence by n levels """
```

Comment:
I'll fill out this docstring soon.

Reply:
Cool.  Don't forget to also point to `concat(seq)`, which flattens a sequence one level.